setup.py

```diff
@@ -35,7 +35,7 @@ def get_package_data(root_dir):
 tests_require = ['coverage', 'jupyter', 'mypy', 'types-redis', 'types-python-dateutil', 'pytest']
 
 install_requires = [
-    'apscheduler', 'attrs', 'dulwich', 'IPython', 'jupyter', 'lmfit', 'matplotlib',
+    'apscheduler', 'attrs', 'dulwich', 'IPython', 'jupyter', 'lmfit', 'matplotlib<3.8.0',
```

Comment:
@bkreynen Restricting matplotlib does not work well with other package requirements. Why did you pin matplotlib? In the builds I could find https://app.travis-ci.com/github/QuTech-Delft/qtt/builds/265966621 which has some mypy errors. 

Can we fix the mypy errors (or just ignore them, sometimes mypy cannot handle the annotations properly)

Reply:
I haven't finished working on this yet. I've marked the PR as draft again. I temporarily pinned matplotlib just to check that the new version of matplotlib was the cause of these errors.